File: objc/Project1/Project1/ViewController.m
//
//  ViewController.m
//  Project1
//
//  Created by Jinwoo Kim on 2/6/21.
//

#import "ViewController.h"

@implementation ViewController

- (void)setup {
    self.categories = @[@"Airplanes", @"Beaches", @"Bridges", @"Cats", @"Cities", @"Dogs", @"Earth", @"Forests", @"Galaxies", @"Landmarks", @"Mountains", @"People", @"Roads", @"Sports", @"Sunets"];
}

- (void)viewDidLoad {
    [self setup];
    [super viewDidLoad];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.categories.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
    cell.textLabel.text = self.categories[indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    ImageViewController *vc = [self.storyboard instantiateViewControllerWithIdentifier:@"Images"];
    if (vc) {
        vc.catetory = self.categories[indexPath.row];
        [self presentViewController:vc animated:YES completion:nil];
    }
}

@end

File: swift/Project1/Project1/ViewController.swift
//
//  ViewController.swift
//  Project1
//
//  Created by Jinwoo Kim on 2/3/21.
//

import UIKit

class ViewController: UIViewController {
    var categories = ["Airplanes", "Beaches", "Bridges", "Cats", "Cities", "Dogs", "Earth", "Forests", "Galaxies", "Landmarks", "Mountains", "People", "Roads", "Sports", "Sunets"]
}

extension ViewController: UITableViewDataSource, UITableViewDelegate {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return categories.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
        cell.textLabel?.text = categories[indexPath.row]
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        guard let vc = storyboard?.instantiateViewController(identifier: "Images") as? ImagesViewController else { return }
        vc.category = categories[indexPath.row]
        present(vc, animated: true)
    }
}
